### ISSUE TITLE
Schematic Editor: reposition junction dots in a senseful way when wires are rearranged

### ISSUE THREAD

**Issue by Schorsch99** [LibrePCB/LibrePCB#1294]
##### SUMMARY
When rearranging wires, junction dots remain in the same position, even if another position makes more sense.

##### SOLUTION
Move junction dot in a senseful way (see KiCad).

![grafik](https://github.com/LibrePCB/LibrePCB/assets/27017311/5f0088a8-5c55-4138-a321-8f9f4db7f0a7)

**Comment by ubruhin** [comment]
I agree, and in general wires should automatically be cleaned up further - not only rearranging junctions, but also removing unnecessary waypoints (e.g. two straight segments into the same direction should be combined).
